libs/util/http.client.js

```diff
@@ -211,6 +211,9 @@ function io(url, options) {
         resp.responseText = body;
 
         if (err) {
+            if (err.message === 'XMLHttpRequest timeout') {
+                err = new Error('XMLHttpRequest timeout: failed to load ' + resp.url + ' in ' + options.timeout + 'ms');
+            }
```

Comment:
@lo1tuma  `resp.url` is assigned into `err.url`, see Line 219 (3 lines down).  And `err.timeout = options.timeout` is also set when status code is 0 or 408.    Will you be able to use those fields for debugging and reporting?  Thanks!